pkg/volume/csi/csi_plugin.go

```diff
@@ -164,7 +165,9 @@ func (p *csiPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.S
 		},
 	}
 
-	return volume.NewSpecFromPersistentVolume(pv, false), nil
+	readOnly, _ := strconv.ParseBool(volData[volDataKey.readOnly])
```

Comment:
If parsing fails, the method should return an error to inform operator that intended readOnly value could not be applied.